src/hooks/useTitle.ts

```diff
@@ -126,49 +203,69 @@ interface MessageWithValues {
 }
 
-const COLONY_HOME_ROUTE_WITHOUT_WILDCARD = COLONY_HOME_ROUTE.slice(0, -2);
-
 const routeMessages: Record<string, MessageDescriptor> = {
   [CREATE_COLONY_ROUTE]: MSG.createColony,
-  [CREATE_USER_ROUTE]: MSG.createUser,
-  [USER_EDIT_PROFILE_ROUTE]: MSG.editProfile,
-  [NOT_FOUND_ROUTE]: MSG.notFound,
-  [LANDING_PAGE_ROUTE]: MSG.landing,
+  [CREATE_PROFILE_ROUTE]: MSG.createProfile,
+  [`${USER_HOME_ROUTE}/${USER_EDIT_PROFILE_ROUTE}`]: MSG.editProfile,
+  [`${USER_HOME_ROUTE}/${USER_PREFERENCES_ROUTE}`]: MSG.editPreferences,
+  [`${USER_HOME_ROUTE}/${USER_ADVANCED_ROUTE}`]: MSG.advancedSettings,
+
+  '/': MSG.landing,
   [COLONY_HOME_ROUTE]: MSG.colonyHome,
-  [COLONY_EVENTS_ROUTE]: MSG.colonyEvents,
-  [COLONY_FUNDING_ROUTE]: MSG.colonyFunds,
-  [`${COLONY_HOME_ROUTE_WITHOUT_WILDCARD}${COLONY_EXTENSIONS_ROUTE}`]:
-    MSG.colonyExtensions,
-  [`${COLONY_HOME_ROUTE_WITHOUT_WILDCARD}${COLONY_EXTENSION_DETAILS_ROUTE}`]:
+  [`${COLONY_HOME_ROUTE}${COLONY_EVENTS_ROUTE}`]: MSG.colonyEvents,
+  [`${COLONY_HOME_ROUTE}${COLONY_FUNDING_ROUTE}`]: MSG.colonyFunds,
+  [`${COLONY_HOME_ROUTE}${COLONY_EXTENSIONS_ROUTE}`]: MSG.colonyExtensions,
+  [`${COLONY_HOME_ROUTE}${COLONY_EXTENSION_DETAILS_ROUTE}`]:
     MSG.colonyExtensionDetails,
-  [`${COLONY_HOME_ROUTE_WITHOUT_WILDCARD}${COLONY_EXTENSION_DETAILS_SETUP_ROUTE}`]:
+  [`${COLONY_HOME_ROUTE}${COLONY_EXTENSION_DETAILS_SETUP_ROUTE}`]:
     MSG.colonyExtensionSetup,
   [USER_ROUTE]: MSG.userProfile,
-  [ACTIONS_PAGE_ROUTE]: MSG.transactionDetails,
-  [COLONY_MEMBERS_ROUTE]: MSG.colonyMembers,
-  '/': MSG.fallbackTitle,
+  [ACTIONS_QUERY_STRING]: MSG.transactionDetails,
+  [`${COLONY_HOME_ROUTE}${COLONY_MEMBERS_ROUTE}`]: MSG.colonyMembers,
+  [`${COLONY_HOME_ROUTE}${COLONY_CONTRIBUTORS_ROUTE}`]: MSG.colonyContributors,
+  [`${COLONY_HOME_ROUTE}${COLONY_FOLLOWERS_ROUTE}`]: MSG.colonyFollowers,
+  [`${COLONY_HOME_ROUTE}${COLONY_VERIFIED_ROUTE}`]: MSG.colonyVerified,
+  [`${COLONY_HOME_ROUTE}${COLONY_TEAMS_ROUTE}`]: MSG.colonyTeams,
+  [`${COLONY_HOME_ROUTE}${COLONY_DETAILS_ROUTE}`]: MSG.colonyDetails,
+  [`${COLONY_HOME_ROUTE}${COLONY_REPUTATION_ROUTE}`]: MSG.colonyReputation,
+  [`${COLONY_HOME_ROUTE}${COLONY_PERMISSIONS_ROUTE}`]: MSG.colonyPermissions,
+  [`${COLONY_HOME_ROUTE}${COLONY_INTEGRATIONS_ROUTE}`]: MSG.colonyIntegrations,
+  [`${COLONY_HOME_ROUTE}${COLONY_INCORPORATION_ROUTE}`]:
+    MSG.colonyIncorporation,
+  [`${COLONY_HOME_ROUTE}${COLONY_ADVANCED_ROUTE}`]: MSG.colonyAdvanced,
+
+  [NOT_FOUND_ROUTE]: MSG.notFound,
 };
 
 const allRoutes = Object.keys(routeMessages);
 
-const getMessageAndValues = (locationPath: string): MessageWithValues => {
+const getMessageAndValues = (
+  locationPath: string,
+  search: string,
+): MessageWithValues => {
+  // If route includes tx search query
+  if (search && /\?tx=[a-zA-F0-9]{66}/.test(search)) {
+    return {
+      msg: routeMessages[ACTIONS_QUERY_STRING],
+      values: { transactionHash: search.split('=')[1] },
+    };
+  }
```

Comment:
This comes back to my previous question about why we're using a query string for the tx (action / motion) route.

As I'm really scared of relying on regex for this filter

Reply:
I have to say, I am, too. Again, I wonder if there might be something from react-router that can help here.